.github/workflows/build_macos.yml

```diff
@@ -123,57 +128,35 @@ jobs:
           key: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.deployment-target }}-${{ matrix.vcpkg-triplet }}-${{ steps.current-date.outputs.stamp }}
           restore-keys: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.deployment-target }}-${{ matrix.vcpkg-triplet }}-
 
-      - name: install dependencies
+      - name: setup ccache
         run: |
-          brew install ccache
-          # add ccamke to PATH - see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions
-          echo "/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
-          if [[ "${{ matrix.system-portaudio }}" != "false" ]]; then brew install portaudio; fi
-          if [[ "${{ matrix.build-fftw }}" != "true" ]]; then brew install fftw; fi
-
-          # install universal versions of homebrew libraries
-          if [[ "${{ matrix.cmake-architectures }}" != "x86_64" ]]; then
-              UNIVERSAL_PACKAGES=
-              if [[ "${{ matrix.build-libsndfile }}" != "true" ]]; then UNIVERSAL_PACKAGES="libsndfile $UNIVERSAL_PACKAGES"; fi
-              if [[ "${{ matrix.build-readline }}" != "true" ]]; then UNIVERSAL_PACKAGES="readline $UNIVERSAL_PACKAGES"; fi
-              if [[ "${{ matrix.build-fftw }}" != "true" ]]; then UNIVERSAL_PACKAGES="fftw $UNIVERSAL_PACKAGES"; fi
-              if [[ "${{ matrix.system-portaudio }}" != "false" ]]; then UNIVERSAL_PACKAGES="portaudio $UNIVERSAL_PACKAGES"; fi
-              echo "Downloading script for creating universal binaries from homebrew packages"
-              curl --no-progress-meter -O -J -L https://gist.githubusercontent.com/dyfer/6c83905d4593750105897e51e87ec345/raw/98e073d755cc3ca8426e20c30f51e9bac2bc9858/brew-install-universal.sh
-              chmod +x brew-install-universal.sh
-              ./brew-install-universal.sh $UNIVERSAL_PACKAGES
-          fi
+          brew install ccache --quiet
+          # add ccache to PATH
+          echo "`brew --prefix ccache`/libexec" >> $GITHUB_PATH
 
-      - name: install libsndfile # to make it compatible with older OSes (lower deployment target)
-        if: matrix.build-libsndfile == true && matrix.vcpkg-triplet
+      - name: setup vcpkg # vcpkg is not automatically installed in newer macOS runner images
+        if: matrix.vcpkg-packages && matrix.vcpkg-triplet
         run: |
-          brew unlink libsndfile libvorbis libogg flac opus || true
-          vcpkg install libsndfile --triplet="${{ matrix.vcpkg-triplet }}" --overlay-triplets="$GITHUB_WORKSPACE/vcpkg/triplets"
+          if [[ -z "${VCPKG_INSTALLATION_ROOT}" ]]; then
```

Comment:
why don't we move the if condition in the shell script into the if condition of the step?

Reply:
Because the build matrix doesn't have access to environment variables of the runner. I thought of this, but alas that doesn't work.

Reply:
There seems something like this possible https://stackoverflow.com/a/74386472 but it is questionable if this is easier to follow then the shell if condition.
Let's keep it this way then :)

Reply:
Yeah, absolutely it's possible to "export" that value from a previous step, but that definitely introduces more noise than this is worth. Direct "if <envVar>" would be great, but when that's not possible, what we have here is clean enough I think.